File: TextAdventure/src/TextAdventure.java
import java.util.Scanner;
import java.util.Random;

public class TextAdventure {
    public static void main(String[] args) {
        int playerLevel = 1;
        Scanner playerInput = new Scanner(System.in);
        System.out.println("What is your character's name?");
        String playerName = playerInput.nextLine();
        Random generator = new Random();
        int playerHealth = generator.nextInt(100);

        System.out.println(playerName + " is level " + playerLevel + " and has " + playerHealth + " health.");
        System.out.println("Before you is a large cave mouth. Darkness is before you. Dare you enter the Cave of Wonders?");

        System.out.println("1. Enter the cave");
        System.out.println("2. Run away");

        int playerChoice = playerInput.nextInt();
        if (playerChoice == 1) {
            //Player enters cave
            playerLevel++;
            System.out.println(playerName + " entered the cave. Congrats! You leveled up to level " + playerLevel + "!");

            System.out.println("You don't get far before you stumble across a guard.");

            System.out.println("1. Sneak past the guard");
            System.out.println("2. Fight the guard");
            System.out.println("3. Talk to the guard");
            boolean
            playerChoice = playerInput.nextInt();
            if (playerChoice == 1) {
                System.out.println("You successfully sneak past the guard.");
                System.out.println("");
            } else if (playerChoice == 2){
                System.out.println("You fight the guard, but he is too strong and well armed to be overpowered.\nComrade! You are dead!");
                System.exit(0);
            }

            System.out.println("You leave the cavern with the chest and continue your journey");
            System.out.println("Up ahead the air grows warm and wet. You hear snoring.");
            System.out.println("Peeking around a corner you see dragon sleeping in the room ahead.");
            System.out.println("You move to leave, but the dragon awakes before you can!");

            System.out.println("1. Fight the Dragon.");
            System.out.println("2. Run for your life.");
                playerChoice = playerInput.nextInt();
                    if (playerChoice == 1) {
                        //Player fights dragon
                        System.out.println("Adrenaline courses through your body and you fight the dragon.");
                        System.out.println("You slay the mighty beast!");
                        System.out.println("Perhaps the leviathan was guarding something.");

                        System.out.println("1. Investigate the lair.");
                        System.out.println("2. Leave the lair.");
                        boolean GoldKey = false;
                        if (playerChoice == 1) {
                            //Player investigates lair
                            GoldKey = true;
                            System.out.println("Beneath the creature's carcass you find a golden key.");

                            System.out.println("You stumble upon a chest.");
                            System.out.println("1. Attempt to open the chest with your key.");
                            System.out.println("2. Move on.");
                            System.out.println("3. Leave the cave and go to sleep.");
                                if (playerChoice == 1) {
                                    System.out.println("Your key gets stuck in the chest and won't come out.");
                                    System.out.println("You move onto the next room and are confronted by a goblin.\nHe asks if you have the key.\nWhen you say no, he stabs you to death.\nComrade! You are dead!");
                                    System.exit(0);
                                } else if (playerChoice == 2) {
                                    System.out.println("You move onto the next room and are confronted by a goblin.\nHe asks if you have the key.\nHe lets you pass when you show it to him.");
                                } else {
                                    System.out.println("You decide that you have had enough adventures for one day and return home to sleep");
                                    System.exit(0);
                                }
                        } else {
                            //Player does not investigate lair
                            System.out.println("You decide that you have had enough adventures for one day and return home to sleep");
                            System.exit(0);
                        }
                    } else {
                        //Player runs
                        System.out.println("As you flee and go to sleep, the dragon returns to its slumber.");
                        System.exit(0);
                    }
                } else {
                    //Player doesn't get Magic Sword
                    System.out.println("Wary of a trap, you ignore the chest.");

                    System.out.println("You leave the cavern with the chest and continue your journey");
                    System.out.println("Up ahead the air grows warm and wet. You hear snoring.");
                    System.out.println("Peeking around a corner you see dragon sleeping in the room ahead.");
                    System.out.println("You move to leave, but the dragon awakes before you can!");

                    System.out.println("1. Fight the Dragon.");
                    System.out.println("2. Run for your life.");
                    playerChoice = playerInput.nextInt();
                if (playerChoice == 1) {
                    //Player fights dragon
                        System.out.println("Adrenaline courses through your body and you fight the dragon.");
                        System.out.println("Blows rain down on the beast's body but you cannot kill the dragon and it eats you!");
                        System.out.println("Comrade! You are dead!");
                        System.exit(0);
                    } else {
                        //Player runs
                        System.out.println("As you flee and go to sleep, the dragon returns to its slumber.");
                        System.exit(0);
                    }
                }

            } else {
                //Player runs from cave
                System.out.println(playerName + " runs home and sleeps.");
                System.exit(0);
            }

        }
    }
    //MSword code
    System.out.println("You come into a natural cavern and find a treasure chest.");
                    boolean hasMSword = false;
                    System.out.println("1. Open the chest.");
                    System.out.println("2. Ignore the chest, it must be a trap.");
                    playerChoice = playerInput.nextInt();

File: TextAdventure.py
#Type "TextAdventure()" into the Thonny shell to begin your adventure.
from random import randint

def TextAdventure():
    playerName = input("""What is your character's name?
    Enter your name: """)
    playerHealth = 100
    playerChoice = 0
    print(playerName + " has " + str(playerHealth) + " health.")

    print("Before you is the mouth of a large, dark cave.")
    print("1. Enter the cave.")
    print("2. Run away.")
    playerChoice = input("""What will you do?
Enter an option: """)
    
    if playerChoice == '1':
        #Player enters the cave.
        print("You don't get far before you stuble upon a guard.")
        print("1. Sneak past the guard.")
        print("2. Fight the guard.")
        print("3. Talk to the guard.")
        playerChoice = input("""What will you do?
Enter an option: """)
        
        if playerChoice == '1':
            print("You successfully sneak past the guard.")
            print("You come into a cavern with a chest at its center.")
            print("1. Open the chest.")
            print("2. Ignore the chest, it must be a trap.")
            playerChoice = input("""What will you do?
Enter an option: """)
                
            if playerChoice == '1':
                print("You open the chest and find a sword inside. It is quite worn, but still seems to be sharp.")
                hasSword = True
                    
            elif playerChoice == '2':
                hasSword = False
                
            print("""You move down the passage at the other enterance. It leads into tunnel and another chamber,
this one home to a pack of wolves. A few of them turns, snarl at you, and begin to creep forward.""")
            print("1. Take on the wolves.")
            print("2. Run for your life.")
            enemies = randint(5, 15)
            enemyHealth = 10
            playerChoice = input("""What will you do?
Enter an option: """)
                
            if playerChoice == '1':
                if hasSword == True:
                    while True:
                        playerHealth -= randint(3, 9)
                        enemyHealth -= randint(4, 10)
                        if enemyHealth < 1:
                            enemies -= 1
                            
                        print("You have " + str(playerHealth) + " health.")
                        print("The wolf has " + str(enemyHealth) + " health. There are " + str(enemies) + " other wolves left.")
                        if enemyHealth < 1:
                            enemyHealth = 10
                        
                        if enemies == 0 or playerHealth < 1:
                            break
                        
                    if playerHealth < 1:
                        print("The wolves tear you to pieces. " + playerName + " is dead.")
                        
                    if enemies == 0:
                        print("You have defeated all of the wolves. " + playerName + """ decides that that is enough adventures for one
day and goes home to sleep.""")
                            
                if hasSword == False:
                    print("The wolves tear you to pieces. " + playerName + " is dead.")
                    
            elif playerChoice == '2':
                print("You escape from the cave, go home, and rest.")
                
        elif playerChoice == '2':
            print("You fight the guard, but he is too strong and well-armed to be overpowered. " +
playerName + " is dead.")
            
        elif playerChoice == '3':
            print("The guard warns you to turn back or be killed.")
            print("1. Fight the guard.")
            print("2. Leave the cave.")
            playerChoice = input("""What will you do?
Enter an option: """)
                
            if playerChoice == '1':
                print("The guard defeats you. " + playerName + " is dead.")
                    
            elif playerChoice == '2':
                print(playerName + " returns home and goes to sleep.")
                    
    elif playerChoice == '2':
        print(playerName + " runs home and goes to sleep.")